include/range/v3/view/unique.hpp

```diff
@@ -33,27 +33,44 @@ namespace ranges
         {
             struct unique_fn
             {
-                template<typename Rng>
+            private:
+                friend view_access;
+                template<typename C,
+                        CONCEPT_REQUIRES_(!Range<C>())>
+                static auto bind(unique_fn unique, C pred)
+                RANGES_DECLTYPE_AUTO_RETURN
+                (
+                    std::bind(unique, std::placeholders::_1, protect(std::move(pred)))
+                )
+
+            public:
+                template<typename Rng, typename C = equal_to,
+                        typename I = iterator_t<Rng>>
                 using Concept = meta::and_<
                     ForwardRange<Rng>,
-                    EqualityComparable<range_value_type_t<Rng>>>;
+                    IndirectRelation<C, I>>;
 
-                template<typename Rng, CONCEPT_REQUIRES_(Concept<Rng>())>
-                unique_view<all_t<Rng>> operator()(Rng && rng) const
```

Comment:
I *think* these are the only uses of `unique_view`. If so, we can remove the definition of the alias in `range_fwd.hpp`.